### DISCUSSION TITLE
Matomo and Lychee use

### DISCUSSION THREAD

**Discussion by bussekas** [LycheeOrg/Lychee#1847] [Q&A] [answered]
Hi
I like to know where i should put https://matomo.org/ analytics code into Lychee gallery?

Urmo

**Comment by ildyria** [comment]
If you are talking about a snippet like this:
```html
<!-- Matomo -->
<script type="text/javascript">
  var _paq = window._paq = window._paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//{$MATOMO_URL}/";
    _paq.push(['setTrackerUrl', u+'matomo.php']);
    _paq.push(['setSiteId', {$IDSITE}]);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
<!-- End Matomo Code -->
```
You can take the script part and insert it into the Configurable JS part in the settings.
Otherwise you may need to hack a bit the templates (see https://github.com/LycheeOrg/Lychee/blob/master/resources/views/layouts/app.blade.php )

**Comment by bussekas** [comment]
why i ask it because i add code yesterday and tested on phone and another site and did not see any visit.
https://foto.urmo.ee/ˇ
and code:
```html
<html>
<body>
<!--StartFragment-->

<!-- Matomo -->
--
  | <script>
  | var _paq = window._paq = window._paq \|\| [];
  | /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
  | _paq.push(['trackPageView']);
  | _paq.push(['enableLinkTracking']);
  | (function() {
  | var u="//matomo.urmo.ee/";
  | _paq.push(['setTrackerUrl', u+'matomo.php']);
  | _paq.push(['setSiteId', '3']);
  | var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
  | g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
  | })();
  | </script>
  | <!-- End Matomo Code -->

<!--EndFragment-->
</body>
</html>
```

into frontend.blade.php file.

**Reply by ildyria** [reply]
indeed frontend.blade.php is the right place.
Also check the security policy. Check the console.log of your webbrowser.
By default Lychee is forbidding inline JS.

**Reply by bussekas** [reply]
but what could be alternative way to get understand visits.

**Reply by ildyria** [reply]
Creating and adding a middleware logger directly in Lychee/Laravel.

**Comment by ildyria** [comment, marked as answer]
https://lycheeorg.github.io/docs/external_tracking.html thanks to [timo-reymann](https://github.com/LycheeOrg/Lychee/commits?author=timo-reymann)
